syntax = "proto3";

service OrderBook {
  // Get the whole order book.
  rpc FetchOrderBook(FetchOrderBookRequest) returns (FetchOrderBookResponse) {}
}

message FetchOrderBookRequest {
}

enum OrderType {
  UNKNOWN = 0;
  ASK = 1;
  BID = 2;
}

message Order {
  OrderType order_type = 1;
  float price = 2;
  float quantity = 3;
}

message FetchOrderBookResponse {
  repeated Order orders = 1;
}